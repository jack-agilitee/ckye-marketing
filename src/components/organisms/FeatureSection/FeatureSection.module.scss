// FeatureSection.module.scss
@import '@/styles/variables';
@import '@/styles/mixins';

.featureSection {
  // Mobile first - default styles
  display: flex;
  flex-direction: column;
  gap: $spacing-4xl; // 64px gap between elements
  
  // Colors
  background-color: $color-background; // #08090a
  color: $color-text-primary; // #f7f8f8
  
  // Typography base
  font-family: $font-family-inter;
  
  // Mobile padding: 40px horizontal, 64px vertical
  padding: $spacing-4xl $spacing-2xl; // 64px 40px
  
  // Tablet and Desktop: change to row layout
  @include breakpoint(small) {
    flex-direction: row;
    padding: $spacing-4xl; // 64px all around
    align-items: center;
  }
  
  // Layout modifier for image position
  &--imageRight {
    @include breakpoint(small) {
      .featureSection__imageWrapper {
        order: 2;
      }
      
      .featureSection__content {
        order: 1;
      }
    }
  }
  
  &--imageLeft {
    @include breakpoint(small) {
      .featureSection__imageWrapper {
        order: 1;
      }
      
      .featureSection__content {
        order: 2;
      }
    }
  }
  
  // Content section - 50% width on desktop/tablet
  &__content {
    display: flex;
    flex-direction: column;
    gap: $spacing-lg; // 24px between heading and body
    width: 100%; // Full width on mobile
    
    @include breakpoint(small) {
      width: 50%; // 50% width on tablet/desktop
    }
  }
  
  // Heading styles
  &__heading {
    color: $color-text-primary; // #f7f8f8
    font-family: $font-family-inter;
    font-weight: $font-weight-medium;
    line-height: $line-height-feature-heading; // 1.2
    margin: 0;
    
    // Mobile: 32px, -0.64px letter spacing
    font-size: $font-size-h3; // 32px (2rem)
    letter-spacing: $letter-spacing-feature-h1-mobile; // -0.64px
    
    // Tablet: 40px, -0.8px letter spacing
    @include breakpoint(medium) {
      font-size: $font-size-h2; // 40px (2.5rem)
      letter-spacing: $letter-spacing-feature-h1-tablet; // -0.8px
    }
    
    // Desktop: 56px, -1.12px letter spacing
    @include breakpoint(large) {
      font-size: $font-size-h1; // 56px (3.5rem)
      letter-spacing: $letter-spacing-feature-h1-desktop; // -1.12px
    }
  }
  
  // Body text styles
  &__body {
    color: $color-text-secondary; // #8b8f97
    font-family: $font-family-inter;
    font-weight: $font-weight-regular;
    line-height: $line-height-feature-body; // 1.4
    margin: 0;
    
    // Mobile: 16px, -0.08px letter spacing
    font-size: $font-size-base; // 16px (1rem)
    letter-spacing: $letter-spacing-feature-body-mobile; // -0.08px
    
    // Tablet: 18px, -0.09px letter spacing
    @include breakpoint(medium) {
      font-size: $font-size-md; // 18px (1.125rem)
      letter-spacing: $letter-spacing-feature-body-tablet; // -0.09px
    }
    
    // Desktop: 24px, -0.12px letter spacing
    @include breakpoint(large) {
      font-size: $font-size-xl; // 24px (1.5rem)
      letter-spacing: $letter-spacing-feature-body-desktop; // -0.12px
    }
  }
  
  // Image wrapper - 50% width on desktop/tablet, scales based on width
  &__imageWrapper {
    position: relative;
    border-radius: $border-radius-2xl; // 16px
    overflow: hidden;
    width: 100%; // Full width on mobile
    aspect-ratio: 544 / 432; // Maintains aspect ratio
    
    @include breakpoint(small) {
      width: 50%; // 50% width on tablet/desktop
      display: flex;
      align-items: center; // Vertically centers the image
    }
  }
  
  // Image styles for Next.js Image component
  &__image {
    border-radius: $border-radius-2xl; // 16px
    object-fit: cover;
  }
}