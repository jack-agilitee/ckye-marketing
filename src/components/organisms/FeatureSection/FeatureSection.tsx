import Image from 'next/image';
import styles from './FeatureSection.module.scss';

interface FeatureSectionProps {
  heading: string;
  bodyText: string;
  image: {
    src: string;
    alt: string;
  };
  layout: 'image-left' | 'image-right';
  className?: string;
}

const FeatureSection: React.FC<FeatureSectionProps> = ({
  heading,
  bodyText,
  image,
  layout,
  className
}) => {
  const layoutClass = layout === 'image-left' 
    ? styles['featureSection--imageLeft'] 
    : styles['featureSection--imageRight'];
  
  return (
    <section className={`${styles.featureSection} ${layoutClass} ${className || ''}`}>
      <div className={styles['featureSection__imageWrapper']}>
        <Image
          src={image.src}
          alt={image.alt}
          fill
          className={styles['featureSection__image']}
          priority={false}
        />
      </div>
      
      <div className={styles['featureSection__content']}>
        <h2 className={styles['featureSection__heading']}>{heading}</h2>
        <p className={styles['featureSection__body']}>{bodyText}</p>
      </div>
    </section>
  );
};

export default FeatureSection;