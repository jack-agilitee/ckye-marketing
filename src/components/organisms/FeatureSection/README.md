# FeatureSection

## Overview
The FeatureSection component is a flexible content display organism that presents information with an accompanying image in two distinct layout variations. It's designed to create engaging content sections with optimal visual hierarchy and responsive behavior across all device sizes.

## Figma Reference
- **Design URL**: [FeatureSection Component](https://figma.com/file/...)
- **Last Updated**: 2025-01-09
- **Designer**: CKYE Design Team

## Installation
```tsx
import FeatureSection from '@/components/organisms/FeatureSection/FeatureSection';
```

## Basic Usage
```tsx
import FeatureSection from '@/components/organisms/FeatureSection/FeatureSection';

function Example() {
  return (
    <FeatureSection
      heading="Innovative Solutions"
      bodyText="Discover cutting-edge technologies that transform your business operations and drive sustainable growth."
      image={{
        src: "/images/feature-innovation.jpg",
        alt: "Modern technology solutions in action"
      }}
      layout="image-left"
    />
  );
}
```

## Props

| Prop | Type | Default | Required | Description |
|------|------|---------|----------|-------------|
| `heading` | `string` | - | Yes | The main heading text displayed in the content section |
| `bodyText` | `string` | - | Yes | The body text that provides detailed description |
| `image` | `{ src: string, alt: string }` | - | Yes | Image object containing source URL and alt text |
| `layout` | `'image-left' \| 'image-right'` | - | Yes | Controls the layout direction and image position |
| `className` | `string` | - | No | Additional CSS classes for custom styling |

## Examples

### Image Left Layout
```tsx
<FeatureSection
  heading="Advanced Analytics"
  bodyText="Leverage powerful data insights to make informed decisions and optimize your business performance across all channels."
  image={{
    src: "/images/analytics-dashboard.jpg",
    alt: "Analytics dashboard showing business metrics"
  }}
  layout="image-left"
/>
```

### Image Right Layout
```tsx
<FeatureSection
  heading="Seamless Integration"
  bodyText="Connect all your business tools and workflows in one unified platform designed for efficiency and scalability."
  image={{
    src: "/images/integration-workflow.jpg",
    alt: "Workflow integration diagram"
  }}
  layout="image-right"
/>
```

### With Custom Styling
```tsx
<FeatureSection
  heading="Custom Solutions"
  bodyText="Tailored solutions that adapt to your unique business requirements and industry-specific challenges."
  image={{
    src: "/images/custom-solutions.jpg",
    alt: "Custom business solutions interface"
  }}
  layout="image-left"
  className="custom-feature-section"
/>
```

### Full Implementation Example
```tsx
import FeatureSection from '@/components/organisms/FeatureSection/FeatureSection';

export default function FeaturesPage() {
  const features = [
    {
      heading: "Real-time Collaboration",
      bodyText: "Work together seamlessly with your team using our advanced real-time collaboration tools and shared workspaces.",
      image: {
        src: "/images/collaboration-team.jpg",
        alt: "Team collaborating in real-time on projects"
      },
      layout: "image-left" as const
    },
    {
      heading: "Enterprise Security",
      bodyText: "Enterprise-grade security measures protect your data with end-to-end encryption and compliance standards.",
      image: {
        src: "/images/security-shield.jpg",
        alt: "Security shield protecting enterprise data"
      },
      layout: "image-right" as const
    }
  ];

  return (
    <main>
      {features.map((feature, index) => (
        <FeatureSection
          key={index}
          heading={feature.heading}
          bodyText={feature.bodyText}
          image={feature.image}
          layout={feature.layout}
        />
      ))}
    </main>
  );
}
```

## Layout Variants

### Image Left (`image-left`)
- **Mobile**: Image appears above content (stacked vertically)
- **Tablet/Desktop**: Image positioned on the left side of the content
- **Use Case**: Ideal for introducing new features or primary content

### Image Right (`image-right`)
- **Mobile**: Image appears above content (stacked vertically)
- **Tablet/Desktop**: Image positioned on the right side of the content
- **Use Case**: Perfect for secondary features or alternating content sections

## Responsive Behavior

### Mobile (< 768px)
- **Layout**: Vertical stack with image above content
- **Image**: 100% width with aspect ratio 544:432
- **Typography**: 
  - Heading: 32px, -0.64px letter-spacing
  - Body: 16px, -0.08px letter-spacing
- **Spacing**: 40px horizontal padding, 64px vertical padding

### Tablet (768px - 1024px)
- **Layout**: Horizontal with image and content side by side
- **Image**: Fixed dimensions 304px × 241px
- **Typography**:
  - Heading: 40px, -0.8px letter-spacing
  - Body: 18px, -0.09px letter-spacing
- **Spacing**: 64px padding on all sides

### Desktop (> 1024px)
- **Layout**: Horizontal with image and content side by side
- **Image**: Fixed dimensions 470px × 374px
- **Typography**:
  - Heading: 56px, -1.12px letter-spacing
  - Body: 24px, -0.12px letter-spacing
- **Spacing**: 64px padding on all sides

## Accessibility

### Semantic HTML
- Uses semantic `<section>` element for proper document structure
- Heading hierarchy with `<h2>` for proper content organization
- Paragraph `<p>` element for body text content

### Image Accessibility
- **Required Alt Text**: All images must include descriptive alt text
- **Next.js Image**: Uses optimized Next.js Image component for performance
- **Object Fit**: Images use `object-fit: cover` for consistent aspect ratios

### Keyboard Navigation
- Component is non-interactive, so no keyboard navigation required
- Content is fully accessible to screen readers
- Proper reading order maintained across all layouts

### ARIA Properties
```tsx
// Example with enhanced accessibility
<FeatureSection
  heading="Accessible Feature"
  bodyText="This feature section includes proper ARIA labeling for screen readers."
  image={{
    src: "/images/accessible-feature.jpg",
    alt: "Detailed description of the feature visualization including key elements and context"
  }}
  layout="image-left"
/>
```

### Screen Reader Support
- Content follows logical reading order
- Image alt text provides meaningful context
- Typography hierarchy is clearly defined
- Color contrast meets WCAG 2.1 AA standards

## Testing

### Test IDs
```tsx
<FeatureSection
  heading="Test Feature"
  bodyText="Feature section for testing purposes"
  image={{
    src: "/test-image.jpg",
    alt: "Test image description"
  }}
  layout="image-left"
  className="test-feature-section"
  data-testid="feature-section"
/>
```

### Testing Examples
```tsx
import { render, screen } from '@testing-library/react';
import FeatureSection from '@/components/organisms/FeatureSection/FeatureSection';

describe('FeatureSection', () => {
  const mockProps = {
    heading: 'Test Heading',
    bodyText: 'Test body text content',
    image: {
      src: '/test-image.jpg',
      alt: 'Test image alt text'
    },
    layout: 'image-left' as const
  };

  test('renders with correct content', () => {
    render(<FeatureSection {...mockProps} />);
    
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent('Test Heading');
    expect(screen.getByText('Test body text content')).toBeInTheDocument();
    expect(screen.getByAltText('Test image alt text')).toBeInTheDocument();
  });

  test('applies correct layout class', () => {
    render(<FeatureSection {...mockProps} layout="image-right" />);
    
    const section = screen.getByRole('region');
    expect(section).toHaveClass('image-right');
  });

  test('accepts custom className', () => {
    render(<FeatureSection {...mockProps} className="custom-class" />);
    
    const section = screen.getByRole('region');
    expect(section).toHaveClass('custom-class');
  });
});
```

## Performance Considerations

### Bundle Size
- **Component**: ~2.1 KB (minified)
- **With styles**: ~3.8 KB (minified)
- **Dependencies**: Next.js Image component

### Optimization Tips
- **Image Optimization**: Uses Next.js Image component for automatic optimization
- **Lazy Loading**: Images load when entering viewport (Next.js default)
- **Responsive Images**: Different image sizes served based on screen size
- **CSS Modules**: Scoped styles prevent style conflicts and enable tree shaking

### Next.js Image Configuration
```tsx
// Recommended image sizes for optimal performance
const imageProps = {
  src: "/images/feature.jpg",
  alt: "Feature description",
  sizes: "(max-width: 768px) 100vw, (max-width: 1024px) 304px, 470px"
};
```

## Content Guidelines

### Heading Best Practices
- **Length**: Keep headings concise (2-8 words ideal)
- **Clarity**: Use action-oriented language
- **Hierarchy**: Ensure logical content flow
- **SEO**: Include relevant keywords naturally

### Body Text Guidelines
- **Length**: 1-3 sentences for optimal readability
- **Tone**: Match brand voice and audience
- **Value**: Clearly communicate benefits
- **Scanability**: Use clear, descriptive language

### Image Requirements
- **Format**: JPG, PNG, or WebP
- **Resolution**: Minimum 940px width for desktop display
- **Aspect Ratio**: Maintain consistent ratios across sections
- **Alt Text**: Descriptive and contextual (not just filename)

## Related Components
- [ContentSection](/src/components/organisms/ContentSection) - Text-only content sections
- [HeroSection](/src/components/organisms/HeroSection) - Primary page headers
- [ImageGallery](/src/components/organisms/ImageGallery) - Multi-image displays

## Changelog

### Version 1.0.0 (2025-01-09)
- Initial release with image-left and image-right layouts
- Responsive design for mobile, tablet, and desktop
- Next.js Image component integration
- SCSS modules with BEM methodology
- Full accessibility compliance
- Comprehensive TypeScript interface