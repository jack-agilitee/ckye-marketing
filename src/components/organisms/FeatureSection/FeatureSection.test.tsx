import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FeatureSection from './FeatureSection';

describe('FeatureSection', () => {
  // Setup
  const defaultProps = {
    heading: 'Test Feature Heading',
    bodyText: 'This is the test body text for the feature section component.',
    image: {
      src: '/test-image.jpg',
      alt: 'Test image description',
    },
    layout: 'image-left' as const,
  };

  const renderComponent = (props = {}) => {
    return render(<FeatureSection {...defaultProps} {...props} />);
  };

  // Rendering Tests
  describe('Rendering', () => {
    it('should render without crashing', () => {
      const { container } = renderComponent();
      const section = container.querySelector('section');
      expect(section).toBeInTheDocument();
    });

    it('should render as a section element', () => {
      const { container } = renderComponent();
      expect(container.firstChild?.nodeName).toBe('SECTION');
    });

    it('should have proper semantic structure', () => {
      const { container } = renderComponent();
      const section = container.querySelector('section');
      expect(section).toBeInTheDocument();
      expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
      expect(screen.getByRole('img')).toBeInTheDocument();
    });
  });

  // Content Rendering Tests
  describe('Content Rendering', () => {
    it('should display the heading text', () => {
      renderComponent({ heading: 'Custom Heading' });
      expect(screen.getByRole('heading', { level: 2, name: 'Custom Heading' })).toBeInTheDocument();
    });

    it('should display the body text', () => {
      renderComponent({ bodyText: 'Custom body text content' });
      expect(screen.getByText('Custom body text content')).toBeInTheDocument();
    });

    it('should render heading as h2 element', () => {
      renderComponent();
      const heading = screen.getByRole('heading', { level: 2 });
      expect(heading.tagName).toBe('H2');
    });

    it('should render body text as paragraph element', () => {
      const { container } = renderComponent();
      const paragraph = container.querySelector('p');
      expect(paragraph).toBeInTheDocument();
      expect(paragraph).toHaveTextContent(defaultProps.bodyText);
    });
  });

  // Image Rendering Tests
  describe('Image Rendering', () => {
    it('should render image with correct src', () => {
      renderComponent({ image: { src: '/custom-image.jpg', alt: 'Custom alt text' } });
      const image = screen.getByRole('img');
      expect(image).toHaveAttribute('src', '/custom-image.jpg');
    });

    it('should render image with correct alt text', () => {
      renderComponent({ image: { src: '/test.jpg', alt: 'Descriptive alt text' } });
      expect(screen.getByAltText('Descriptive alt text')).toBeInTheDocument();
    });

    it('should render image without fill attribute in DOM', () => {
      renderComponent();
      const image = screen.getByRole('img');
      // The fill prop is handled by Next.js and shouldn't appear in the DOM
      expect(image).not.toHaveAttribute('fill');
    });

    it('should render image without priority attribute in DOM', () => {
      renderComponent();
      const image = screen.getByRole('img');
      // The priority prop is handled by Next.js and shouldn't appear in the DOM
      expect(image).not.toHaveAttribute('priority');
    });
  });

  // Layout Tests
  describe('Layout Variations', () => {
    it('should apply image-left layout class', () => {
      const { container } = renderComponent({ layout: 'image-left' });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('image-left');
    });

    it('should apply image-right layout class', () => {
      const { container } = renderComponent({ layout: 'image-right' });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('image-right');
    });

    it('should always have base featureSection class', () => {
      const { container } = renderComponent();
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('featureSection');
    });

    it('should handle layout prop changes', () => {
      const { container, rerender } = renderComponent({ layout: 'image-left' });
      let section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('image-left');
      expect(section).not.toHaveClass('image-right');

      rerender(<FeatureSection {...defaultProps} layout="image-right" />);
      section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('image-right');
      expect(section).not.toHaveClass('image-left');
    });
  });

  // CSS Classes Tests
  describe('CSS Classes', () => {
    it('should apply custom className when provided', () => {
      const { container } = renderComponent({ className: 'custom-class' });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('custom-class');
    });

    it('should apply multiple classes correctly', () => {
      const { container } = renderComponent({ 
        className: 'custom-class another-class',
        layout: 'image-right'
      });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('featureSection');
      expect(section).toHaveClass('image-right');
      expect(section).toHaveClass('custom-class');
      expect(section).toHaveClass('another-class');
    });

    it('should handle undefined className gracefully', () => {
      const { container } = renderComponent({ className: undefined });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('featureSection');
      expect(section.className).not.toContain('undefined');
    });

    it('should handle empty string className', () => {
      const { container } = renderComponent({ className: '' });
      const section = container.firstChild as HTMLElement;
      expect(section).toHaveClass('featureSection');
    });

    it('should apply correct CSS module classes to child elements', () => {
      const { container } = renderComponent();
      
      // Check container div
      expect(container.querySelector('.container')).toBeInTheDocument();
      
      // Check image wrapper
      expect(container.querySelector('.imageWrapper')).toBeInTheDocument();
      
      // Check content wrapper
      expect(container.querySelector('.content')).toBeInTheDocument();
      
      // Check heading class
      expect(container.querySelector('.heading')).toBeInTheDocument();
      
      // Check body text class
      expect(container.querySelector('.bodyText')).toBeInTheDocument();
      
      // Check image class
      expect(container.querySelector('.image')).toBeInTheDocument();
    });
  });

  // Props Validation Tests
  describe('Props Validation', () => {
    it('should handle empty heading', () => {
      renderComponent({ heading: '' });
      const heading = screen.getByRole('heading', { level: 2 });
      expect(heading).toHaveTextContent('');
    });

    it('should handle empty body text', () => {
      const { container } = renderComponent({ bodyText: '' });
      const paragraph = container.querySelector('p');
      expect(paragraph?.tagName).toBe('P');
      expect(paragraph?.textContent).toBe('');
    });

    it('should handle special characters in text content', () => {
      const specialText = 'Special chars: !@#$%^&*()_+-={}[]|\\:";\'<>?,./';
      renderComponent({ 
        heading: specialText,
        bodyText: specialText 
      });
      expect(screen.getByRole('heading', { name: specialText })).toBeInTheDocument();
      expect(screen.getAllByText(specialText)).toHaveLength(2); // heading and body
    });

    it('should handle long text content', () => {
      const longText = 'This is a very long text '.repeat(50);
      const { container } = renderComponent({ 
        heading: longText,
        bodyText: longText 
      });
      const heading = screen.getByRole('heading');
      const paragraph = container.querySelector('p');
      
      expect(heading.textContent).toBe(longText);
      expect(paragraph?.textContent).toBe(longText);
      expect(heading.textContent?.length).toBeGreaterThan(100);
    });

    it('should handle HTML entities in text', () => {
      const textWithEntities = 'AT&T - Johnson & Johnson - Ben & Jerry\'s';
      renderComponent({ 
        heading: textWithEntities,
        bodyText: textWithEntities 
      });
      expect(screen.getByRole('heading', { name: textWithEntities })).toBeInTheDocument();
      expect(screen.getAllByText(textWithEntities)).toHaveLength(2);
    });
  });

  // Image Props Tests
  describe('Image Props', () => {
    it('should handle different image extensions', () => {
      const imageTypes = ['.jpg', '.jpeg', '.png', '.webp', '.svg'];
      
      imageTypes.forEach((ext, index) => {
        const { container } = renderComponent({ 
          image: { 
            src: `/test-image${ext}`, 
            alt: `Test ${ext} image` 
          }
        });
        const image = container.querySelector('img');
        expect(image).toHaveAttribute('src', `/test-image${ext}`);
        expect(image).toHaveAttribute('alt', `Test ${ext} image`);
      });
    });

    it('should handle external image URLs', () => {
      renderComponent({ 
        image: { 
          src: 'https://example.com/image.jpg', 
          alt: 'External image' 
        }
      });
      expect(screen.getByRole('img')).toHaveAttribute('src', 'https://example.com/image.jpg');
    });

    it('should handle empty alt text', () => {
      const { container } = renderComponent({ 
        image: { 
          src: '/test.jpg', 
          alt: '' 
        }
      });
      const image = container.querySelector('img');
      expect(image).toHaveAttribute('alt', '');
    });

    it('should handle alt text with special characters', () => {
      const altText = 'Image with "quotes" & ampersands < > symbols';
      renderComponent({ 
        image: { 
          src: '/test.jpg', 
          alt: altText 
        }
      });
      expect(screen.getByAltText(altText)).toBeInTheDocument();
    });
  });

  // Structure and DOM Tests
  describe('DOM Structure', () => {
    it('should have correct DOM hierarchy', () => {
      const { container } = renderComponent();
      
      // section > div.container > [div.imageWrapper, div.content]
      const section = container.querySelector('section');
      const containerDiv = section?.querySelector('.container');
      const imageWrapper = containerDiv?.querySelector('.imageWrapper');
      const contentDiv = containerDiv?.querySelector('.content');
      const heading = contentDiv?.querySelector('h2.heading');
      const bodyText = contentDiv?.querySelector('p.bodyText');
      const image = imageWrapper?.querySelector('img.image');

      expect(section).toBeInTheDocument();
      expect(containerDiv).toBeInTheDocument();
      expect(imageWrapper).toBeInTheDocument();
      expect(contentDiv).toBeInTheDocument();
      expect(heading).toBeInTheDocument();
      expect(bodyText).toBeInTheDocument();
      expect(image).toBeInTheDocument();
    });

    it('should maintain proper parent-child relationships', () => {
      const { container } = renderComponent();
      
      const section = container.querySelector('section');
      const containerDiv = container.querySelector('.container');
      const imageWrapper = container.querySelector('.imageWrapper');
      const contentDiv = container.querySelector('.content');

      expect(containerDiv?.parentElement).toBe(section);
      expect(imageWrapper?.parentElement).toBe(containerDiv);
      expect(contentDiv?.parentElement).toBe(containerDiv);
    });
  });

  // Integration Tests
  describe('Component Integration', () => {
    it('should work with all props provided', () => {
      const allProps = {
        heading: 'Integration Test Heading',
        bodyText: 'Integration test body text with comprehensive content.',
        image: {
          src: '/integration-test.jpg',
          alt: 'Integration test image'
        },
        layout: 'image-right' as const,
        className: 'integration-test-class'
      };

      const { container } = renderComponent(allProps);

      expect(screen.getByRole('heading', { name: allProps.heading })).toBeInTheDocument();
      expect(screen.getByText(allProps.bodyText)).toBeInTheDocument();
      expect(screen.getByAltText(allProps.image.alt)).toBeInTheDocument();
      expect(screen.getByRole('img')).toHaveAttribute('src', allProps.image.src);
      
      const section = container.querySelector('section');
      expect(section).toHaveClass('featureSection');
      expect(section).toHaveClass('image-right');
      expect(section).toHaveClass('integration-test-class');
    });

    it('should work with minimal required props', () => {
      const minimalProps = {
        heading: 'Min',
        bodyText: 'Min body',
        image: {
          src: '/min.jpg',
          alt: 'Min'
        },
        layout: 'image-left' as const
      };

      const { container } = renderComponent(minimalProps);

      expect(screen.getByRole('heading', { name: 'Min' })).toBeInTheDocument();
      expect(screen.getByText('Min body')).toBeInTheDocument();
      expect(screen.getByAltText('Min')).toBeInTheDocument();
      
      const section = container.querySelector('section');
      expect(section).toHaveClass('featureSection');
      expect(section).toHaveClass('image-left');
    });
  });

  // Snapshot Tests
  describe('Snapshot Tests', () => {
    it('should match snapshot with image-left layout', () => {
      const { container } = renderComponent({ layout: 'image-left' });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with image-right layout', () => {
      const { container } = renderComponent({ layout: 'image-right' });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with custom className', () => {
      const { container } = renderComponent({ 
        className: 'custom-feature-section',
        layout: 'image-left'
      });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with different content', () => {
      const { container } = renderComponent({
        heading: 'Different Heading for Snapshot',
        bodyText: 'Different body text content for comprehensive snapshot testing.',
        image: {
          src: '/different-image.jpg',
          alt: 'Different image for snapshot'
        },
        layout: 'image-right'
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});