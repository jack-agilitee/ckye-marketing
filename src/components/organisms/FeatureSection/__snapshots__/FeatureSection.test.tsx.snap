// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FeatureSection Snapshot Tests should match snapshot with custom className 1`] = `
<section
  class="featureSection image-left custom-feature-section"
>
  <div
    class="container"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="Test image description"
        class="image"
        src="/test-image.jpg"
      />
    </div>
    <div
      class="content"
    >
      <h2
        class="heading"
      >
        Test Feature Heading
      </h2>
      <p
        class="bodyText"
      >
        This is the test body text for the feature section component.
      </p>
    </div>
  </div>
</section>
`;

exports[`FeatureSection Snapshot Tests should match snapshot with different content 1`] = `
<section
  class="featureSection image-right "
>
  <div
    class="container"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="Different image for snapshot"
        class="image"
        src="/different-image.jpg"
      />
    </div>
    <div
      class="content"
    >
      <h2
        class="heading"
      >
        Different Heading for Snapshot
      </h2>
      <p
        class="bodyText"
      >
        Different body text content for comprehensive snapshot testing.
      </p>
    </div>
  </div>
</section>
`;

exports[`FeatureSection Snapshot Tests should match snapshot with image-left layout 1`] = `
<section
  class="featureSection image-left "
>
  <div
    class="container"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="Test image description"
        class="image"
        src="/test-image.jpg"
      />
    </div>
    <div
      class="content"
    >
      <h2
        class="heading"
      >
        Test Feature Heading
      </h2>
      <p
        class="bodyText"
      >
        This is the test body text for the feature section component.
      </p>
    </div>
  </div>
</section>
`;

exports[`FeatureSection Snapshot Tests should match snapshot with image-right layout 1`] = `
<section
  class="featureSection image-right "
>
  <div
    class="container"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="Test image description"
        class="image"
        src="/test-image.jpg"
      />
    </div>
    <div
      class="content"
    >
      <h2
        class="heading"
      >
        Test Feature Heading
      </h2>
      <p
        class="bodyText"
      >
        This is the test body text for the feature section component.
      </p>
    </div>
  </div>
</section>
`;
